File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        prev_idx = 0

        for i in range(len(nums)):
            if nums[i] != 0:
                nums[prev_idx] , nums[i] = nums[i], nums[prev_idx]
                prev_idx += 1
        

File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int* ptr;
        int hold;
        ptr = &nums[0];

        for(int i=0; i < nums.size(); i++) {
            if (nums[i] != 0) {
                hold = *ptr;

                *ptr = nums[i];
                nums[i] = hold;
                ptr++;
            }
        }
        
        
    }
};